1-js/06-advanced-functions/05-global-object/article.md

```diff
@@ -1,155 +1,87 @@
+```
 
-    ```html run
-    <script type="module">
-      var x = 5;
+No navegador, funções e variáveis globais declaradas com `var` (não `let/const`!) tornam-se propriedade do objeto global:
 
-      alert(window.x); // undefined
-    </script>
-    ```
+```js run untrusted refresh
+var gVar = 5;
 
-- Two modules that do not see variables of each other:
+alert(window.gVar); // 5 (se torna uma propriedade do objeto global)
+```
 
-    ```html run
-    <script type="module">
-      let x = 5;
-    </script>
+Por favor, não confie nisso! Esse comportamento existe por motivos de compatibilidade. Scripts modernos usam [JavaScript modules](info:modules) onde tal coisa não acontece.
 
-    <script type="module">
-      alert(window.x); // undefined
-      alert(x); // Error: undeclared variable
-    </script>
-    ```
+Se usássemos `let`, isso não aconteceria:
 
-- And, the last minor thing, the top-level value of `this` in a module is `undefined` (why should it be `window` anyway?):
+```js run untrusted refresh
+let gLet = 5;
 
-    ```html run
-    <script type="module">
-      alert(this); // undefined
-    </script>
-    ```
+alert(window.gLet); // undefined (não se torna uma propriedade do objeto global)
+```
 
-**Using `<script type="module">` fixes the design flaw of the language by separating top-level scope from `window`.**
+Se um valor é tão importante que você gostaria de deixá-lo disponível globalmente, escreva-o diretamente como uma propriedade.
 
-We'll cover more features of modules later, in the chapter [](info:modules).
+```js run
+*!*
+// tornando as informações de current user global, para permitir que todos os script as acessem
+window.currentUser = {
+  name: "John"
+};
+*/!*
+
+// em outro lugar no código
+alert(currentUser.name);  // John
+
+// ou, se tivermos um variável local com o nome "currentUser"
+// a obtenha explicitamente do window (seguro!)
+alert(window.currentUser.name); // John
+```
 
-## Valid uses of the global object
+Dito isto, o uso de variáveis globais é geralmente desencorajado. Deve haver o menor número possível de variáveis globais. O design do código onde uma função recebe variáveis de entrada e produz certos resultados é mais claro, menos propenso a erros e mais fácil de testar do que se usar varáveis externas ou globais.
 
-1. Using global variables is generally discouraged. There should be as few global variables as possible, but if we need to make something globally visible, we may want to put it into `window` (or `global` in Node.js).
+## Usando para polyfills
 
-    Here we put the information about the current user into a global object, to be accessible from all other scripts:
+Usamos o objeto global para testar o suporte aos recursos modernos da linguagem.
 
-    ```js run
-    // explicitly assign it to `window`
-    window.currentUser = {
-      name: "John",
-      age: 30
-    };
+Por exemplo, testar se o objeto interno `Promise` existe (ele não existe em navegadores antigos):
```

Comment:
I think the best translation for built-in would be "nativo" as in https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects